src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -240,7 +274,7 @@ def array_access_parallelisable(self, loop_variable, var_info):
                               "does not depend on the loop "
```

Comment:
Not from this PR but this error clearly means that the variable is being written to. However, I don't think there's any mention of this earlier in this routine. I think the routine name probably needs tweaking along with the docstring.

Reply:
Not sure - the test if it's written to is in the same function, so we can't just call it 'check_written_array'. I've added a list of warnings that might be raised in this function. Does this help?

Reply:
Ah, I must have missed that earlier check. Please just add a comment saying that we've already checked for this earlier.

Reply:
Done.

Reply:
Great, thanks.